// @awesome-cordova-plugins/diagnostic/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { AwesomeCordovaNativePlugin, cordova, cordovaPropertyGet, cordovaPropertySet } from '@awesome-cordova-plugins/core';
var DiagnosticOriginal = /** @class */ (function (_super) {
    __extends(DiagnosticOriginal, _super);
    function DiagnosticOriginal() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.permission = {
            ACCEPT_HANDOVER: 'ACCEPT_HANDOVER',
            ACCESS_BACKGROUND_LOCATION: 'ACCESS_BACKGROUND_LOCATION',
            ACCESS_COARSE_LOCATION: 'ACCESS_COARSE_LOCATION',
            ACCESS_FINE_LOCATION: 'ACCESS_FINE_LOCATION',
            ACCESS_MEDIA_LOCATION: 'ACCESS_MEDIA_LOCATION',
            ACTIVITY_RECOGNITION: 'ACTIVITY_RECOGNITION',
            ADD_VOICEMAIL: 'ADD_VOICEMAIL',
            ANSWER_PHONE_CALLS: 'ANSWER_PHONE_CALLS',
            BLUETOOTH_ADVERTISE: 'BLUETOOTH_ADVERTISE',
            BLUETOOTH_CONNECT: 'BLUETOOTH_CONNECT',
            BLUETOOTH_SCAN: 'BLUETOOTH_SCAN',
            BODY_SENSORS: 'BODY_SENSORS',
            BODY_SENSORS_BACKGROUND: 'BODY_SENSORS_BACKGROUND',
            CALL_PHONE: 'CALL_PHONE',
            CAMERA: 'CAMERA',
            GET_ACCOUNTS: 'GET_ACCOUNTS',
            NEARBY_WIFI_DEVICES: 'NEARBY_WIFI_DEVICES',
            POST_NOTIFICATIONS: 'POST_NOTIFICATIONS',
            PROCESS_OUTGOING_CALLS: 'PROCESS_OUTGOING_CALLS',
            READ_CALENDAR: 'READ_CALENDAR',
            READ_CALL_LOG: 'READ_CALL_LOG',
            READ_CONTACTS: 'READ_CONTACTS',
            READ_EXTERNAL_STORAGE: 'READ_EXTERNAL_STORAGE',
            READ_MEDIA_AUDIO: 'READ_MEDIA_AUDIO',
            READ_MEDIA_IMAGES: 'READ_MEDIA_IMAGES',
            READ_MEDIA_VIDEO: 'READ_MEDIA_VIDEO',
            READ_PHONE_NUMBERS: 'READ_PHONE_NUMBERS',
            READ_PHONE_STATE: 'READ_PHONE_STATE',
            READ_SMS: 'READ_SMS',
            RECEIVE_MMS: 'RECEIVE_MMS',
            RECEIVE_SMS: 'RECEIVE_SMS',
            RECEIVE_WAP_PUSH: 'RECEIVE_WAP_PUSH',
            RECORD_AUDIO: 'RECORD_AUDIO',
            SEND_SMS: 'SEND_SMS',
            USE_SIP: 'USE_SIP',
            UWB_RANGING: 'UWB_RANGING',
            WRITE_CALENDAR: 'WRITE_CALENDAR',
            WRITE_CALL_LOG: 'WRITE_CALL_LOG',
            WRITE_CONTACTS: 'WRITE_CONTACTS',
            WRITE_EXTERNAL_STORAGE: 'WRITE_EXTERNAL_STORAGE',
        };
        _this.locationAuthorizationMode = {
            ALWAYS: 'always',
            WHEN_IN_USE: 'when_in_use',
        };
        /**
         * Location accuracy authorization
         */
        _this.locationAccuracyAuthorization = {
            FULL: 'full',
            REDUCED: 'reduced',
        };
        _this.permissionGroups = {
            CALENDAR: ['READ_CALENDAR', 'WRITE_CALENDAR'],
            CAMERA: ['CAMERA'],
            CONTACTS: ['READ_CONTACTS', 'WRITE_CONTACTS', 'GET_ACCOUNTS'],
            LOCATION: ['ACCESS_FINE_LOCATION', 'ACCESS_COARSE_LOCATION'],
            MICROPHONE: ['RECORD_AUDIO'],
            PHONE: [
                'READ_PHONE_STATE',
                'CALL_PHONE',
                'ADD_VOICEMAIL',
                'USE_SIP',
                'PROCESS_OUTGOING_CALLS',
                'READ_CALL_LOG',
                'WRITE_CALL_LOG',
            ],
            SENSORS: ['BODY_SENSORS'],
            SMS: ['SEND_SMS', 'RECEIVE_SMS', 'READ_SMS', 'RECEIVE_WAP_PUSH', 'RECEIVE_MMS'],
            STORAGE: ['READ_EXTERNAL_STORAGE', 'WRITE_EXTERNAL_STORAGE'],
            NEARBY_DEVICES: ["BLUETOOTH_ADVERTISE", "BLUETOOTH_SCAN", "BLUETOOTH_CONNECT"],
        };
        _this.locationMode = {
            HIGH_ACCURACY: 'high_accuracy',
            DEVICE_ONLY: 'device_only',
            BATTERY_SAVING: 'battery_saving',
            LOCATION_OFF: 'location_off',
        };
        _this.bluetoothState = {
            UNKNOWN: 'unknown',
            RESETTING: 'resetting',
            UNSUPPORTED: 'unsupported',
            UNAUTHORIZED: 'unauthorized',
            POWERED_OFF: 'powered_off',
            POWERED_ON: 'powered_on',
            POWERING_OFF: 'powering_off',
            POWERING_ON: 'powering_on',
        };
        /**
         * Access to the photo library (iOS 14+)
         *
         * ADD_ONLY - can add to but not read from Photo Library
         * READ_WRITE - can both add to and read from Photo Library
         *
         */
        _this.photoLibraryAccessLevel = {
            ADD_ONLY: 'add_only',
            READ_WRITE: 'read_write',
        };
        return _this;
    }
    DiagnosticOriginal.prototype.isLocationAvailable = function () { return cordova(this, "isLocationAvailable", {}, arguments); };
    DiagnosticOriginal.prototype.isWifiAvailable = function () { return cordova(this, "isWifiAvailable", {}, arguments); };
    DiagnosticOriginal.prototype.isCameraAvailable = function (externalStorage) { return cordova(this, "isCameraAvailable", { "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.isBluetoothAvailable = function () { return cordova(this, "isBluetoothAvailable", {}, arguments); };
    DiagnosticOriginal.prototype.switchToLocationSettings = function () { return cordova(this, "switchToLocationSettings", { "sync": true, "platforms": ["Android", "Windows 10", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.switchToMobileDataSettings = function () { return cordova(this, "switchToMobileDataSettings", { "sync": true, "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.switchToBluetoothSettings = function () { return cordova(this, "switchToBluetoothSettings", { "sync": true, "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.switchToWifiSettings = function () { return cordova(this, "switchToWifiSettings", { "sync": true, "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.isWifiEnabled = function () { return cordova(this, "isWifiEnabled", { "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.setWifiState = function (state) { return cordova(this, "setWifiState", { "callbackOrder": "reverse", "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.setBluetoothState = function (state) { return cordova(this, "setBluetoothState", { "callbackOrder": "reverse", "platforms": ["Android", "Windows 10"] }, arguments); };
    DiagnosticOriginal.prototype.enableDebug = function () { return cordova(this, "enableDebug", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isLocationEnabled = function () { return cordova(this, "isLocationEnabled", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isLocationAuthorized = function () { return cordova(this, "isLocationAuthorized", {}, arguments); };
    DiagnosticOriginal.prototype.getLocationAuthorizationStatus = function () { return cordova(this, "getLocationAuthorizationStatus", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getLocationAuthorizationStatuses = function () { return cordova(this, "getLocationAuthorizationStatuses", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestLocationAuthorization = function (mode, accuracy) { return cordova(this, "requestLocationAuthorization", { "platforms": ["Android", "iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.isCameraPresent = function () { return cordova(this, "isCameraPresent", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isCameraAuthorized = function (externalStorage) { return cordova(this, "isCameraAuthorized", { "platforms": ["Android", "iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.getCameraAuthorizationStatus = function (externalStorage) { return cordova(this, "getCameraAuthorizationStatus", { "platforms": ["Android", "iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.requestCameraAuthorization = function (externalStorage) { return cordova(this, "requestCameraAuthorization", { "platforms": ["Android", "iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.isMicrophoneAuthorized = function () { return cordova(this, "isMicrophoneAuthorized", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getMicrophoneAuthorizationStatus = function () { return cordova(this, "getMicrophoneAuthorizationStatus", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestMicrophoneAuthorization = function () { return cordova(this, "requestMicrophoneAuthorization", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isContactsAuthorized = function () { return cordova(this, "isContactsAuthorized", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getContactsAuthorizationStatus = function () { return cordova(this, "getContactsAuthorizationStatus", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestContactsAuthorization = function () { return cordova(this, "requestContactsAuthorization", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isCalendarAuthorized = function () { return cordova(this, "isCalendarAuthorized", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getCalendarAuthorizationStatus = function () { return cordova(this, "getCalendarAuthorizationStatus", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestCalendarAuthorization = function () { return cordova(this, "requestCalendarAuthorization", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.switchToSettings = function () { return cordova(this, "switchToSettings", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getBluetoothState = function () { return cordova(this, "getBluetoothState", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.registerBluetoothStateChangeHandler = function (handler) { return cordova(this, "registerBluetoothStateChangeHandler", { "platforms": ["Android", "iOS"], "sync": true }, arguments); };
    DiagnosticOriginal.prototype.registerLocationStateChangeHandler = function (handler) { return cordova(this, "registerLocationStateChangeHandler", { "platforms": ["Android", "iOS"], "sync": true }, arguments); };
    DiagnosticOriginal.prototype.getArchitecture = function () { return cordova(this, "getArchitecture", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getCurrentBatteryLevel = function () { return cordova(this, "getCurrentBatteryLevel", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.restart = function (cold) { return cordova(this, "restart", { "platforms": ["Android"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.isGpsLocationAvailable = function () { return cordova(this, "isGpsLocationAvailable", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isGpsLocationEnabled = function () { return cordova(this, "isGpsLocationEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isNetworkLocationAvailable = function () { return cordova(this, "isNetworkLocationAvailable", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isNetworkLocationEnabled = function () { return cordova(this, "isNetworkLocationEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isAirplaneModeEnabled = function () { return cordova(this, "isAirplaneModeEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isMobileDataEnabled = function () { return cordova(this, "isMobileDataEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getLocationMode = function () { return cordova(this, "getLocationMode", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getDeviceOSVersion = function () { return cordova(this, "getDeviceOSVersion", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getBuildOSVersion = function () { return cordova(this, "getBuildOSVersion", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getPermissionAuthorizationStatus = function (permission) { return cordova(this, "getPermissionAuthorizationStatus", { "platforms": ["Android"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.getPermissionsAuthorizationStatus = function (permissions) { return cordova(this, "getPermissionsAuthorizationStatus", { "platforms": ["Android"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.requestRuntimePermission = function (permission) { return cordova(this, "requestRuntimePermission", { "platforms": ["Android"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.requestRuntimePermissions = function (permissions) { return cordova(this, "requestRuntimePermissions", { "platforms": ["Android"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.isRequestingPermission = function () { return cordova(this, "isRequestingPermission", { "sync": true }, arguments); };
    DiagnosticOriginal.prototype.registerPermissionRequestCompleteHandler = function (handler) { return cordova(this, "registerPermissionRequestCompleteHandler", { "sync": true }, arguments); };
    DiagnosticOriginal.prototype.isBluetoothEnabled = function () { return cordova(this, "isBluetoothEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.hasBluetoothSupport = function () { return cordova(this, "hasBluetoothSupport", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.hasBluetoothLESupport = function () { return cordova(this, "hasBluetoothLESupport", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.hasBluetoothLEPeripheralSupport = function () { return cordova(this, "hasBluetoothLEPeripheralSupport", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getBluetoothAuthorizationStatus = function () { return cordova(this, "getBluetoothAuthorizationStatus", { "platforms": ["Android", "iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getBluetoothAuthorizationStatuses = function () { return cordova(this, "getBluetoothAuthorizationStatuses", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isExternalStorageAuthorized = function () { return cordova(this, "isExternalStorageAuthorized", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getExternalStorageAuthorizationStatus = function () { return cordova(this, "getExternalStorageAuthorizationStatus", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.requestExternalStorageAuthorization = function () { return cordova(this, "requestExternalStorageAuthorization", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.getExternalSdCardDetails = function () { return cordova(this, "getExternalSdCardDetails", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.switchToWirelessSettings = function () { return cordova(this, "switchToWirelessSettings", { "platforms": ["Android"], "sync": true }, arguments); };
    DiagnosticOriginal.prototype.switchToNFCSettings = function () { return cordova(this, "switchToNFCSettings", { "platforms": ["Android"], "sync": true }, arguments); };
    DiagnosticOriginal.prototype.isNFCPresent = function () { return cordova(this, "isNFCPresent", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isNFCEnabled = function () { return cordova(this, "isNFCEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isNFCAvailable = function () { return cordova(this, "isNFCAvailable", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.registerNFCStateChangeHandler = function (handler) { return cordova(this, "registerNFCStateChangeHandler", { "platforms": ["Android"], "sync": true }, arguments); };
    DiagnosticOriginal.prototype.isDataRoamingEnabled = function () { return cordova(this, "isDataRoamingEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isADBModeEnabled = function () { return cordova(this, "isADBModeEnabled", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isDeviceRooted = function () { return cordova(this, "isDeviceRooted", { "platforms": ["Android"] }, arguments); };
    DiagnosticOriginal.prototype.isCameraRollAuthorized = function (accessLevel) { return cordova(this, "isCameraRollAuthorized", { "platforms": ["iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.getCameraRollAuthorizationStatus = function (accessLevel) { return cordova(this, "getCameraRollAuthorizationStatus", { "platforms": ["iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.requestCameraRollAuthorization = function (accessLevel) { return cordova(this, "requestCameraRollAuthorization", { "platforms": ["iOS"], "callbackOrder": "reverse" }, arguments); };
    DiagnosticOriginal.prototype.presentLimitedLibraryPicker = function () { return cordova(this, "presentLimitedLibraryPicker", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isRemoteNotificationsEnabled = function () { return cordova(this, "isRemoteNotificationsEnabled", { "platforms": ["iOS", "Android"] }, arguments); };
    DiagnosticOriginal.prototype.isRegisteredForRemoteNotifications = function () { return cordova(this, "isRegisteredForRemoteNotifications", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getRemoteNotificationsAuthorizationStatus = function () { return cordova(this, "getRemoteNotificationsAuthorizationStatus", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestRemoteNotificationsAuthorization = function (types, omitRegistration) { return cordova(this, "requestRemoteNotificationsAuthorization", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getRemoteNotificationTypes = function () { return cordova(this, "getRemoteNotificationTypes", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isRemindersAuthorized = function () { return cordova(this, "isRemindersAuthorized", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getRemindersAuthorizationStatus = function () { return cordova(this, "getRemindersAuthorizationStatus", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestRemindersAuthorization = function () { return cordova(this, "requestRemindersAuthorization", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isBackgroundRefreshAuthorized = function () { return cordova(this, "isBackgroundRefreshAuthorized", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getBackgroundRefreshStatus = function () { return cordova(this, "getBackgroundRefreshStatus", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestBluetoothAuthorization = function () { return cordova(this, "requestBluetoothAuthorization", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isMotionAvailable = function () { return cordova(this, "isMotionAvailable", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.isMotionRequestOutcomeAvailable = function () { return cordova(this, "isMotionRequestOutcomeAvailable", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.requestMotionAuthorization = function () { return cordova(this, "requestMotionAuthorization", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getMotionAuthorizationStatus = function () { return cordova(this, "getMotionAuthorizationStatus", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.getLocationAccuracyAuthorization = function () { return cordova(this, "getLocationAccuracyAuthorization", { "platforms": ["iOS", "Android"] }, arguments); };
    DiagnosticOriginal.prototype.requestTemporaryFullAccuracyAuthorization = function (purpose) { return cordova(this, "requestTemporaryFullAccuracyAuthorization", { "platforms": ["iOS"] }, arguments); };
    DiagnosticOriginal.prototype.registerLocationAccuracyAuthorizationChangeHandler = function (handler) { return cordova(this, "registerLocationAccuracyAuthorizationChangeHandler", { "platforms": ["iOS"], "sync": true }, arguments); };
    Object.defineProperty(DiagnosticOriginal.prototype, "permissionStatus", {
        get: function () { return cordovaPropertyGet(this, "permissionStatus"); },
        set: function (value) { cordovaPropertySet(this, "permissionStatus", value); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiagnosticOriginal.prototype, "NFCState", {
        get: function () { return cordovaPropertyGet(this, "NFCState"); },
        set: function (value) { cordovaPropertySet(this, "NFCState", value); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiagnosticOriginal.prototype, "cpuArchitecture", {
        get: function () { return cordovaPropertyGet(this, "cpuArchitecture"); },
        set: function (value) { cordovaPropertySet(this, "cpuArchitecture", value); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiagnosticOriginal.prototype, "remoteNotificationType", {
        get: function () { return cordovaPropertyGet(this, "remoteNotificationType"); },
        set: function (value) { cordovaPropertySet(this, "remoteNotificationType", value); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(DiagnosticOriginal.prototype, "motionStatus", {
        get: function () { return cordovaPropertyGet(this, "motionStatus"); },
        set: function (value) { cordovaPropertySet(this, "motionStatus", value); },
        enumerable: false,
        configurable: true
    });
    DiagnosticOriginal.pluginName = "Diagnostic";
    DiagnosticOriginal.plugin = "cordova.plugins.diagnostic";
    DiagnosticOriginal.pluginRef = "cordova.plugins.diagnostic";
    DiagnosticOriginal.repo = "https://github.com/dpa99c/cordova-diagnostic-plugin";
    DiagnosticOriginal.platforms = ["Android", "iOS", "Windows"];
    return DiagnosticOriginal;
}(AwesomeCordovaNativePlugin));
var Diagnostic = new DiagnosticOriginal();
export { Diagnostic };
